pkg/resourcecollector/resourcecollector.go

```diff
@@ -0,0 +1,356 @@
+	if err != nil {
+		return false, err
+	}
+
+	switch objectType.GetKind() {
+	case "Service":
+		// Don't migrate the kubernetes service
+		metadata, err := meta.Accessor(object)
+		if err != nil {
+			return false, err
+		}
+		if metadata.GetName() == "kubernetes" {
+			return false, nil
+		}
+	case "PersistentVolumeClaim":
+		metadata, err := meta.Accessor(object)
+		if err != nil {
+			return false, err
+		}
+		pvcName := metadata.GetName()
+		pvc, err := k8s.Instance().GetPersistentVolumeClaim(pvcName, namespace)
+		if err != nil {
+			return false, err
+		}
+		if pvc.Status.Phase != v1.ClaimBound {
+			return false, nil
+		}
+
+		if !r.Driver.OwnsPVC(pvc) {
+			return false, nil
+		}
+		return true, nil
+	case "PersistentVolume":
+		phase, err := collections.GetString(object.UnstructuredContent(), "status.phase")
+		if err != nil {
+			return false, err
+		}
+		if phase != string(v1.ClaimBound) {
+			return false, nil
+		}
+		pvcName, err := collections.GetString(object.UnstructuredContent(), "spec.claimRef.name")
+		if err != nil {
+			return false, err
+		}
+		if pvcName == "" {
+			return false, nil
+		}
+
+		pvcNamespace, err := collections.GetString(object.UnstructuredContent(), "spec.claimRef.namespace")
+		if err != nil {
+			return false, err
+		}
+		if pvcNamespace != namespace {
+			return false, nil
+		}
+
+		pvc, err := k8s.Instance().GetPersistentVolumeClaim(pvcName, pvcNamespace)
+		if err != nil {
+			return false, err
+		}
+		if !r.Driver.OwnsPVC(pvc) {
+			return false, nil
+		}
+
+		if len(pvc.Labels) == 0 && len(labelSelectors) > 0 {
+			return false, nil
+		}
+
+		if !labels.AreLabelsInWhiteList(labels.Set(labelSelectors),
+			labels.Set(pvc.Labels)) {
+			return false, nil
+		}
+		return true, nil
+	case "ClusterRoleBinding":
+		name := metadata.GetName()
+		crb, err := k8s.Instance().GetClusterRoleBinding(name)
+		if err != nil {
+			return false, err
+		}
+		for _, subject := range crb.Subjects {
+			if subject.Namespace == namespace {
+				return true, nil
+			}
+		}
+		return false, nil
+	case "ClusterRole":
+		name := metadata.GetName()
+		crbs, err := k8s.Instance().ListClusterRoleBindings()
+		if err != nil {
+			return false, err
+		}
+		for _, crb := range crbs.Items {
+			if crb.RoleRef.Name == name {
+				for _, subject := range crb.Subjects {
+					if subject.Namespace == namespace {
+						return true, nil
+					}
+				}
+			}
+		}
+		return false, nil
+
+	case "ServiceAccount":
+		// Don't migrate the default service account
+		name := metadata.GetName()
+		if name == "default" {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func (r *ResourceCollector) preparePVResource(
+	object runtime.Unstructured,
+) (runtime.Unstructured, error) {
+	spec, err := collections.GetMap(object.UnstructuredContent(), "spec")
+	if err != nil {
+		return nil, err
+	}
+	delete(spec, "claimRef")
+	delete(spec, "storageClassName")
+
+	return object, nil
+}
+
+func (r *ResourceCollector) prepareServiceResource(
```

Comment:
Add a comment to all the perpareXXXResource functions, as to what they are doing and why.

Reply:
`prepareServiceResource` does exactly one thing and there is a comment inline